noxfile.py

```diff
@@ -49,6 +50,8 @@ def lint(session):
     session.run("ruff", "check", "rustworkx", "retworkx", "setup.py")
     session.run("cargo", "fmt", "--all", "--", "--check", external=True)
     session.run("python", "tools/find_stray_release_notes.py")
+    session.run("typos", "--exclude", "releasenotes")
```

Comment:
I will keep this is `nox -etypos` for two reasons:
* it doesn't need rustworkx to be compiled (just like Black)
* It might make our life easier if we want to have the separate CI pass